frontend/cache.lua

```diff
@@ -197,9 +296,41 @@ function Cache:clear()
     self.current_memsize = 0
 end
 
+-- Terribly crappy workaround: evict half the cache if we appear to be redlining on free RAM...
+function Cache:redlineCheck()
```

Comment:
(I'm not familiar with the word "redline" :/ even on Google it is hard to find what it means.)